test/verify/verify_program.hpp

```diff
@@ -24,20 +24,47 @@
 #ifndef MIGRAPHX_GUARD_AUTO_REGISTER_VERIFY_PROGRAM_HPP
 #define MIGRAPHX_GUARD_AUTO_REGISTER_VERIFY_PROGRAM_HPP
 
+#include <functional>
 #include <migraphx/auto_register.hpp>
 #include <migraphx/program.hpp>
-#include <functional>
+#include <migraphx/compile_options.hpp>
 
 struct program_info
 {
     std::string name;
     std::string section;
     std::function<migraphx::program()> get_program;
+    migraphx::compile_options c_options;
 };
 
 void register_program_info(const program_info& pi);
 const std::vector<program_info>& get_programs();
 
+// Don't have access to rank.hpp from CMake configuration
+template <int N>
+struct rank : rank<N - 1>
+{
+};
+
+template <>
+struct rank<0>
+{
+};
+
+namespace detail {
+template <class T>
+auto get_co(rank<1>, const T& t) -> decltype(t.get_compile_options())
+{
+    return t.get_compile_options();
+}
+
+template <class T>
+auto get_co(rank<0>, const T&) -> decltype(migraphx::compile_options{})
+{
+    return migraphx::compile_options{};
+}
```

Comment:
Just add `get_compile_options` to the `verify_program` base class instead of this. Then you wont need `rank<>` class.